src/dfcx_scrapi/core/scrapi_base.py

```diff
@@ -396,6 +398,51 @@ def get_api_calls_count(self) -> int:
         return sum(self.get_api_calls_details().values())
 
 
+    @staticmethod
+    def _update_kwargs(obj, **kwargs) -> field_mask_pb2.FieldMask:
+        """Create a FieldMask for Environment, Experiment, TestCase, Version."""
+        if kwargs:
+            for key, value in kwargs.items():
+                setattr(obj, key, value)
+            return field_mask_pb2.FieldMask(paths=kwargs.keys())
+        attrs_map = {
+            "Environment": [
+                "name", "display_name", "description", "version_configs",
+                "update_time", "test_cases_config", "webhook_config",
+            ],
+            "Experiment": [
+                "name", "display_name", "description", "state", "definition",
+                "rollout_config", "rollout_state", "rollout_failure_reason",
+                "result", "create_time", "start_time", "end_time",
+                "last_update_time", "experiment_length", "variants_history",
+            ],
+            "TestCase": [
+                "name", "tags", "display_name", "notes", "test_config",
+                "test_case_conversation_turns", "creation_time",
+                "last_test_result",
+            ],
+            "Version": [
+                "name", "display_name", "description", "nlu_settings",
+                "create_time", "state",
+            ],
+        }
+        if isinstance(obj, types.Environment):
+            paths = attrs_map["Environment"]
```

Comment:
@MRyderOC how do these work if I only want to update 1 single kwarg? For example if I only pass in `display_name` but you set `["name", "display_name", "description", "nlu_settings", "create_time", "state",]`.

Does this null out the other data? or only update the 1 kwarg?

Reply:
Well in this method we check for kwargs first and if there is any we use that for updating: https://github.com/GoogleCloudPlatform/dfcx-scrapi/blob/4ec03f79b58653352874181992f7c6a8cf1c10cd/src/dfcx_scrapi/core/scrapi_base.py#L404-L407

Reply:
Discussed offline and understand the process here now! great solution! thanks!